basic-cache/basic-cache-logic/src/in_memory_cache.rs

```diff
@@ -0,0 +1,68 @@
+use std::{collections::HashMap, hash::Hash, sync::Mutex};
+
+use crate::CacheManager;
+
+#[derive(Debug)]
+pub struct HashMapCache<K, V> {
+    // need interior mutability because usual work with DB is done
+    // via immutable reference
+    inner: Mutex<HashMap<K, V>>,
```

Comment:
Maybe RwLock would be better?

Reply:
There's no shared ownership of `Mutex`, and the code is single threaded. No difference I guess?

Reply:
in general case I see what you mean though :)